FirebaseFunctionsSwift/Tests/IntegrationTests.swift

```diff
@@ -0,0 +1,652 @@
+    func testTokenAsync() async throws {
+      // Recreate functions with a token.
+      let functions = FunctionsFake(
+        projectID: "functions-integration-test",
+        region: "us-central1",
+        customDomain: nil,
+        withToken: "token"
+      )
+      functions.useLocalhost()
+
+      let function = functions.httpsCallable(
+        "FCMTokenTest",
+        requestAs: [String: Int].self,
+        responseAs: [String: Int].self
+      )
+
+      let data = try await function.call([:])
+      XCTAssertEqual(data, [:])
+    }
+  #endif
+
+  func testFCMToken() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "FCMTokenTest",
+      requestAs: [String: Int].self,
+      responseAs: [String: Int].self
+    )
+    function.call([:]) { result in
+      do {
+        let data = try result.get()
+        XCTAssertEqual(data, [:])
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 5)
+  }
+
+  #if compiler(>=5.5) && canImport(_Concurrency)
+    @available(iOS 15, tvOS 15, macOS 12, watchOS 8, *)
+    func testFCMTokenAsync() async throws {
+      let function = functions.httpsCallable(
+        "FCMTokenTest",
+        requestAs: [String: Int].self,
+        responseAs: [String: Int].self
+      )
+
+      let data = try await function.call([:])
+      XCTAssertEqual(data, [:])
+    }
+  #endif
+
+  func testNull() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "nullTest",
+      requestAs: Int?.self,
+      responseAs: Int?.self
+    )
+    function.call(nil) { result in
+      do {
+        let data = try result.get()
+        XCTAssertEqual(data, nil)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 5)
+  }
+
+  #if compiler(>=5.5) && canImport(_Concurrency)
+    @available(iOS 15, tvOS 15, macOS 12, watchOS 8, *)
+    func testNullAsync() async throws {
+      let function = functions.httpsCallable(
+        "nullTest",
+        requestAs: Int?.self,
+        responseAs: Int?.self
+      )
+
+      let data = try await function.call(nil)
+      XCTAssertEqual(data, nil)
+    }
+  #endif
+
+  // No parameters to call should be the same as passing nil.
+  // If no parameters are required, then the non-typed API
+  // is more appropriate since it specifically avoids defining
+  // type.
+  //  func testParameterless() {
+  //  }
+  //
+  //
+  func testMissingResult() throws {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable(
+      "missingResultTest",
+      requestAs: Int?.self,
+      responseAs: Int?.self
+    )
+    function.call(nil) { result in
+      do {
+        _ = try result.get()
```

Comment:
Should this also have a `XCTFail("Failed to throw error for missing result")`?

Reply:
Yep. Good catch! Added here and below.